### ISSUE TITLE
gpu leakage

### ISSUE THREAD

**Issue by eeric** [chufengt/iccv19_attribute#36]
in main.py:
the line 196, 256:
output = model(input)
it caused gpu leakage.

**Comment by eeric** [comment]
adding ahead
with torch.no_grad():

**Comment by chufengt** [comment]
Thanks for your notice!

**Comment by eeric** [comment]
my torch==1.5.0, torchvision==0.6.0

**Comment by eeric** [comment]
run successfully in peta dataset as following:

poch: [1][0/357]       Time 2.284 (2.284)      Loss 2.0522 (2.0522)    Accu 0.808 (0.808)
Epoch: [1][100/357]     Time 2.234 (2.115)      Loss 2.2972 (2.1147)    Accu 0.804 (0.808)
Epoch: [1][200/357]     Time 2.103 (2.108)      Loss 2.0128 (2.0976)    Accu 0.834 (0.810)
Epoch: [1][300/357]     Time 2.113 (2.110)      Loss 2.0380 (2.0935)    Accu 0.815 (0.810)
Test: [0/238]   Time 1.064 (1.064)      Loss 2.0559 (2.0559)    Accu 0.798 (0.798)
Test: [100/238] Time 1.041 (1.098)      Loss 2.3746 (2.0712)    Accu 0.781 (0.808)
Test: [200/238] Time 0.964 (1.076)      Loss 2.0761 (2.0695)    Accu 0.811 (0.808)
 * Accu 0.808
====================================================================================================
             Attr               p_true/n_true   p_tol/n_tol     p_pred/n_pred   cur_mA
        # 0: Age16-30           3233\2091       3728\3872       5014\2586       0.70363
        # 1: Age31-45           2026\2681       2514\5086       4431\3169       0.66651
        # 2: Age46-60            284\6182        790\6810        912\6688       0.63364
        # 3: AgeAbove61          329\6707        479\7121        743\6857       0.81435
        # 4: Backpack            644\5352       1444\6156       1448\6152       0.65769
        # 5: CarryingOther       414\5309       1511\6089       1194\6406       0.57295
        # 6: Casual lower       6448\ 192       6523\1077       7333\ 267       0.58339
        # 7: Casual upper       6383\ 181       6468\1132       7334\ 266       0.57338
        # 8: Formal lower        729\4477       1062\6538       2790\4810       0.68560
        # 9: Formal upper        699\4748       1042\6558       2509\5091       0.69741
        #10: Hat                 510\5921        772\6828       1417\6183       0.76389
        #11: Jacket               94\6751        515\7085        428\7172       0.56769
        #12: Jeans              1546\4073       2279\5321       2794\4806       0.72191
        #13: Leather Shoes      1938\2811       2293\5307       4434\3166       0.68743
        #14: Logo                  0\7308        290\7310          2\7598       0.49986
        #15: Long hair           855\5101       1827\5773       1527\6073       0.67579
        #16: Male               4110\ 757       4180\3420       6773\ 827       0.60230
        #17: Messenger Bag      1053\4350       2237\5363       2066\5534       0.64092
        #18: Muffler             237\6807        637\6963        393\7207       0.67483
        #19: No accessory       5631\ 154       5673\1927       7404\ 196       0.53626
        #20: No carrying        1797\2384       2140\5460       4873\2727       0.63817
        #21: Plaid                22\7379        193\7407         50\7550       0.55510
        #22: PlasticBags         123\6946        574\7026        203\7397       0.60145
        #23: Sandals               0\7435        165\7435          0\7600       0.50000
        #24: Shoes              2091\2416       2763\4837       4512\3088       0.62813
        #25: Shorts               72\6959        259\7341        454\7146       0.61298
        #26: Short Sleeve        640\5875       1083\6517       1282\6318       0.74622
        #27: Skirt                 1\7263        337\7263          1\7599       0.50148
        #28: Sneaker            1025\4635       1600\6000       2390\5210       0.70656
        #29: Stripes               0\7469        131\7469          0\7600       0.50000
        #30: Sunglasses            0\7376        224\7376          0\7600       0.50000
        #31: Trousers           3809\ 536       3932\3668       6941\ 659       0.55742
        #32: Tshirt              256\6563        638\6962        655\6945       0.67197
        #33: UpperOther         3047\2087       3456\4144       5104\2496       0.69264
        #34: V-Neck                0\7511         89\7511          0\7600       0.50000
        mA:        0.624901743529178
        Accuracy:  0.5107288602587899
        Precision: 0.5810388345207389
        Recall:    0.7861626936659638
        F1_Score:  0.6682132017175125
====================================================================================================
